internal/ingress/controller/controller.go

```diff
@@ -1329,6 +1329,10 @@ func mergeAlternativeBackend(priUps *ingress.Backend, altUps *ingress.Backend) b
 		}
 	}
 
+	if ing.ParsedAnnotations != nil && ing.ParsedAnnotations.SessionAffinity.CanaryBehavior != "legacy" {
```

Comment:
As ing is also a pointer, it's never enough to check if it's also nil (I know, corner case, but better be safe than sorry!) 

So:

```suggestion
	if ing != nil && ing.ParsedAnnotations != nil && ing.ParsedAnnotations.SessionAffinity.CanaryBehavior != "legacy" {
```

Reply:
I understand this suggestion, however, this method is a private utility used by the `mergeAlternativeBackends` method, which already assumes that `ing` is not null (see line 1350).